### PR TITLE
[PT-671] Update the status endpoint to display the version number

### PR DESCRIPTION
**Pull request by ydrisR**
Use the features from go-common to display the version number on the status route.
Now the status object (from go-common) is always used, no matter whether the status is OK or NOK.

The variables `ReleaseNumber` and `FullCommit` have to be injected at build time using:
`go build -ldflags '-X main.ReleaseNumber=1.0.5 -X main.FullCommit=123456789bla'`

### CODE REVIEW

**Comment by ydrisR** [comment]
**WARNING** do not merge the PR as is. I changed the toml file to reference the relevant branche in go-common ( `feature/PT-727_Enhance_Status_Report`) instead of master.

**Comment by ydrisR** [comment]
Another detail to validate: the status object {code, reason, error, version} is not used for the status url only. It is returned in all error responses by hydrophone (see the last commit to fix signup tests).  
I don't think it is a problem to add the version in these error responses but I'd like a second opinion.

File: Dockerfile [hunk 1]
```diff
@@ -11,12 +11,7 @@ RUN apk --no-cache update && \
 
 RUN go get -u github.com/golang/dep/cmd/dep
 
-RUN  dos2unix build.sh && ./build.sh && \
```
Comment (BigFab) [inline Dockerfile:14]: so we assume we won't have need for "dos2unix" anymore?

**Review by BigFab — COMMENTED** [review]
LGTM although I would have taken the release version number from the CHANGELOG instead of Github

Tests were OK on my side. I tested by running a build.sh then through the docker container. I got the release number I was expecting i.e. retrieved from the git repository

I am waiting for the final commit (to include the "released" go-common lib) before retesting/approving/merging

Good job @ydrisR @CedricPerotto !

File: Dockerfile (hunk 1 above)
Comment (ydrisR) [reply Dockerfile:14]: good question... I hope we won't since we know where the problem was comming from ;)

File: build.sh [hunk 2]
```diff
@@ -3,12 +3,23 @@
...
+else 
+    VERSION_BASE=$(git describe --abbrev=0 --tags 2> /dev/null || echo 'dblp.0.0.0')
+fi
+VERSION_SHORT_COMMIT=$(git rev-parse --short HEAD)
+VERSION_FULL_COMMIT=$(git rev-parse HEAD)
+
+GO_COMMON_PATH="github.com/tidepool-org/go-common"
+	
+echo "Build hydrophone $VERSION_BASE+$VERSION_FULL_COMMIT"
+go build -ldflags "-X $GO_COMMON_PATH/clients/version.ReleaseNumber=$VERSION_BASE \
```
Comment (BigFab) [inline build.sh:19]: Looks like a good practice to run "go mod tidy" to prune the dependencies before building. Does it make sense?
https://github.com/golang/go/wiki/Modules#how-to-prepare-for-a-release

Comment (BigFab) [inline build.sh:19]:
```suggestion
go mod tidy
go build -ldflags "-X $GO_COMMON_PATH/clients/version.ReleaseNumber=$VERSION_BASE \
```

**Review by BigFab — CHANGES_REQUESTED** [review]

**Review by BigFab — APPROVED** [review]
LGTM

**Pull request merged**